lib/timbot.py

```diff
@@ -0,0 +1,131 @@
+
+	def send_message(self, text):
+		self.client.api_call('chat.postMessage', channel=self.channel_id, text=text, as_user=False)
+
+	def send_lunch_time_notification(self):
+		self.send_message("IT IS THE IDEAL LUNCH TIME GO TO LUNCH")
+
+	def send_chat_image(self, image_name, image_url, title, text):
+		if image_name is not None:
+			filepath = os.path.join(self.images_base_dir, image_name)
+
+			try:
+				with open(filepath, 'rb') as f:
+					resp = self.client.api_call(
+						"files.upload",
+						title='sampletitle',
+						file=io.BytesIO(f.read()),
+				)
+			except IOError:
+				print("Failed to open file: the path '{}' does not exist".format(filepath))
+				return
+
+			meta = resp['file']
+			self.client.api_call("files.sharedPublicURL", file=meta['id'])
+			attachments = [{"title": title, "image_url": meta['permalink_public']}]
+		else:
+			attachments = [{"title": title, "image_url": image_url}]
+
+		resp = self.client.api_call("chat.postMessage", channel=self.channel_id, text=text, attachments=attachments, as_user=True)
+
+	def handle_message(self, message):
+		# base case: handle unrecognized input
+		if 'lunch' not in message:
+			self.default_message_handler()
+			return
+
+		if 'where' in message:
+			if (datetime.datetime.today().weekday() == self.weekday_lunch_index) or 'no feast' in message:
+				random.seed(datetime.datetime.now())
+				self.send_message(helpers.choose_lunchcation())
+			else:
+				self.send_message(self.default_lunchcation)
+		elif 'when' in message or 'time' in message:
+			self.send_chat_image("lunchcharts.png", '', 'IdealLunchTimeChart', self.ideal_lunch_time)
+		elif 'what' in message and 'eat' in message:
+			if datetime.datetime.today().weekday() == self.weekday_lunch_index:
+				self.send_message('Its friday enjoy a meal out. Maybe some french toast at pauls?')
+			else:
+				self.send_chat_image('', 'http://cafe.epicureanfeast.com/Clients/8680redhat.jpg', 'Menu', 'May I suggest the chicken sandwich')
+
+	def payload_handler(self, message):
+		# check if the current message was in the form of '@timbot ...'
+		# current problem: message is in the form of "<@ur9ha8kv2> hey"
+		# note: DM history may be handled differently than channel history
+		if message.startswith(self.timbot_user_id):
+			self.handle_message(message)
+		elif 'openstack' in message:
+			self.send_message("I hear Openstack is a career killer")
+		elif 'pong' in message:
+			self.send_message("I\'m in. Best 2 out of 3 games to 7?")
+
+	def run(self):
+		timbot_user_id_striped = self.timbot_user_id.strip('<@>')
+
+		while True:
+			history = self.get_channel_history()
+
+			if datetime.datetime.now().strftime('%H:%M') == self.ideal_lunch_time:
+				self.send_lunch_time_notification()
+
+			# wait until there are messages populated in the history dictionary
+			if 'messages' not in history or len(history['messages']) == 0:
+				time.sleep(1.5)
+
+			for data in history['messages']:
```

Comment:
key 'messages' is not checked

Reply:
It wouldn't need to be checked as we verified there's a message key in the `while` statement above.